model/map/src/main/java/org/keycloak/models/map/authorization/adapter/MapResourceAdapter.java

```diff
@@ -129,6 +132,25 @@ public void setOwnerManagedAccess(boolean ownerManagedAccess) {
     @Override
     public void updateScopes(Set<Scope> scopes) {
         throwExceptionIfReadonly();
+
+        PermissionTicketStore permissionStore = storeFactory.getPermissionTicketStore();
```

Comment:
Would be nice if we could remove everything in a single step. I remember some discussions about this.

Reply:
Removing permissions could be done in a single step, we would need something like `removeByScope` or `removeByAttributes` similarly as we do [searching](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/authorization/MapPermissionTicketStore.java#L186).

However, I am not sure about the policies as they need to be updated, not removed.

Anyway, this would be out of scope for this PR as it is about disabling caching. I will create an issue for addressing this.

Reply:
https://github.com/keycloak/keycloak/issues/10597